apps/passport-client/src/loadInitialState.ts

```diff
@@ -43,20 +44,25 @@ export async function loadInitialState(): Promise<AppState> {
     self &&
     (!self.semaphore_v4_commitment || !self.semaphore_v4_pubkey)
   ) {
-    const semaphoreV4IdentityPCD = findIdentityV4PCD(pcds);
-    if (semaphoreV4IdentityPCD) {
-      await requestUpgradeUserWithV4Commitment(
-        appConfig.zupassServer,
-        await makeUpgradeUserWithV4CommitmentRequest(pcds)
-      );
-    }
+    const identityPCD = v3IdentityToPCD(identityV3);
+
+    await requestUpgradeUserWithV4Commitment(
+      appConfig.zupassServer,
+      await makeUpgradeUserWithV4CommitmentRequest(
+        new PCDCollection(await getPackages(), [identityPCD])
+      )
+    );
+
     const newSelfResponse = await requestUser(
       appConfig.zupassServer,
       self.uuid
     );
+
     if (newSelfResponse.success) {
       self = newSelfResponse.value;
       saveSelf(self);
+    } else {
+      // anything to do here?
```

Comment:
I think this is a valid case in offline mode.  The client should continue using the old self it already had.

Note that I think this synchronous update of `self` during page load might be something to consider eliminating for performance reasons, but that's out of the scope of this PR.